### ISSUE TITLE
Zoom increments are incredibly small

### ISSUE THREAD

**Issue by 0ion9** [luspi/photoqt-dev#55]
.. And there doesn't appear to be a setting to customize this.
I just tested -- average time that I have to hold + to get a small image to 'acceptable' size is 4 seconds, maximum is 8 seconds. It seems to take about 1 second of continuously holding the zoom button for the image to zoom in +100%.

Being able to zoom in/out by 100% at a time (ie. by a single keypress) seems more practical.

**Comment by luspi** [comment]
Hi @0ion9 
Thanks for your feedback. I actually hadn't really noticed how slow the zoom currently is (I agree, it's pretty bad right now). And yes, unfortunately there isn't a setting to change that yet. I will add one though, as already now, the zoom speed is controlled by a single variable, so reading the value of the variable from the settings is a trivial thing to do...
I like the idea of zooming in 100% at a time. I might see if I can add it as an optional shortcut that can be set if desired...

**Comment by luspi** [comment]
Btw, this repo here contains only the legacy code of PhotoQt (v1.2.1). All development will happen in a completely new code basis you can find here: https://gitlab.com/luspi/photoqt

**Comment by 0ion9** [comment]
Re: zoom speed: Now that I think of it, some programs, like GIMP, Inkscape, Mypaint.., instead use exponential zooming (eg. [100%, 300%, 800%.]). This is also an option worth considering. (Obviously one that I favor, but I consider myself biased because I use paint programs a lot.)

Okay. FWIW I got here by googling 'photoqt git' and this was the top result. Then I saw your link which said photoqt-legacy, went there, concluded it was old and came back here.

So I guess I'm saying that message could stand to reference the link you give above. Something like "This (legacy) repository is now on gitlab. However, if you are looking for the current development version of PhotoQT, instead go here (link you gave above)"

**Comment by tpatter2** [comment]
Just curious, what platform are you using 0ion9? I'm using a VM with Ubuntu that doesn't have this issue at all. it takes about a second to zoom from maximum to minimum.

**Comment by 0ion9** [comment]
Arch Linux x86_64.
Do note that the problem is worse the smaller the image is. My tests were mostly made with images < 32x32. For a 512x624 image, 100% change takes about a second. For a 1172x1808 image, zooming from minimum to maximum takes three seconds.

Zooming out, for some reason, is slower than zooming in, so that 1172x1808 image takes four seconds to zoom out to minimum from maximum.  This too is worse for small images -- small images zoom out much slower, and sufficiently small image refuse to zoom out at all. Rounding error related?
Here is a small (18x15) image that exhibits the 'just can't zoom out' problem:
![goblin](https://cloud.githubusercontent.com/assets/115782/10383666/98b59a2a-6e79-11e5-9b9d-9d808e7cf072.png)

**Comment by luspi** [comment]
I'm terribly sorry for only replying now! I honestly thought I had long replied...

I'm also using Arch Linux 64 Bit for developing PhotoQt, and I was also able to observe the issue. PhotoQt does use some sort of exponential zooming. E.g., when zooming in a large image, the zoomsteps get progressively smaller. Now, the problem arising from that is, that the calculations I'm doing for that don't work for images as small as the ones you're using. I didn't think of that use case as I set them up... Also as I was looking at that part of the code, I noticed another bug, that causes the behaviour of zooming out to be slower than zooming in (in short: the zoom steps are only re-calculated when zooming in, zooming out is stuck with whatever was in the respective variables before).

I haven't been able to do much programming for PhotoQt recently, as I had a large programming task to do as part of my research for my studies. However, that has progressed pretty well, so I'll be able to spend a bit more time with PhotoQt now. I'll let you know once I fixed that behaviour.

In regards to which repo to use: You're right, the message could actually be a little misleading. I'll update it! But lets stay here with these issues, no need to duplicate them there. I'll add the zooming bugs to the website (http://photoqt.org/bugs) and simply link to here instead of somewhere on GitLab.
